src/cc2olx/models.py

```diff
@@ -0,0 +1,625 @@
+            if res_filename.suffix == ".html":
+                try:
+                    with open(res_filename) as res_file:
+                        html = res_file.read()
+                except:  # noqa: E722
+                    print(
+                        "Failure reading {!r} from id {}".format(
+                            res_filename, identifier
+                        )
+                    )
+                    raise
+                return "html", {"html": html}
+            else:
+                print("*** Skipping webcontent: {}".format(res_filename))
+                return None, None
+        elif res_type == "imswl_xmlv1p1":
+            tree = filesystem.get_xml_tree(self._res_filename(res["children"][0].href))
+            root = tree.getroot()
+            ns = {"wl": "http://www.imsglobal.org/xsd/imsccv1p1/imswl_v1p1"}
+            title = root.find("wl:title", ns).text
+            url = root.find("wl:url", ns).get("href")
+            return "link", {"href": url, "text": title}
+        elif res_type == "imsbasiclti_xmlv1p0":
+            data = self._parse_lti(res)
+            return "lti", data
+        else:
+            text = "Unimported content: type = {!r}".format(res_type)
+            if "href" in res:
+                text += ", href = {!r}".format(res["href"])
+            print("***", text)
+            return "html", {"html": text}
+
+    def load_manifest_extracted(self):
+        manifest = self._extract()
+        tree = filesystem.get_xml_tree(manifest)
+        root = tree.getroot()
+        self._update_namespaces(root)
+        data = self._parse_manifest(root)
+        self.metadata = data["metadata"]
+        self.organizations = data["organizations"]
+        self.resources = data["resources"]
+        self.resources_by_id = {r["identifier"]: r for r in self.resources}
+        self.version = self.metadata.get("schema", {}).get("version", self.version)
+        return data
+
+    def write_xml(self, text, output_base, output_file):
+        text += "\n"
+        output_path = os.path.join(output_base, output_file)
+        with open(output_path, "w") as output:
+            output.write(text)
+
+    def get_course_xml(self):
+        text = '<course org="{org}" course="{number}" url_name="{run}" />'.format(
+            org=self.get_course_org(),
+            number=self.get_course_number(),
+            run=self.get_course_run(),
+        )
+        return text
+
+    def get_run_xml(self):
+        text = (
+            dedent(
+                """\
+            <course
+                display_name="{title}"
+                language="{language}"
+            >
+            </course>\
+        """
+            )
+            .format(title=self.get_title(), language=self.get_language(), )
+            .strip()
+        )
+        return text
+
+    def get_title(self):
+        # TODO: Choose a better default course title
```

Comment:
Just adding a note of this TODO.  Not sure if it is from current work or copied.

Reply:
These TODO's are from original repo. [Source](https://github.com/openedx/cc2olx/blob/f395393e7e6f661cdeb5e28c838940de00e247ac/src/cc2olx/models.py#L310).

Reply:
Ok so updated in the future refactoring?